src/main/kotlin/io/emeraldpay/dshackle/upstream/forkchoice/PriorityForkChoice.kt

```diff
@@ -22,7 +22,8 @@ class PriorityForkChoice : ForkChoice {
 
     override fun filter(block: BlockContainer): Boolean {
         val curr = head.get()
-        return (curr == null || curr.nodeRating <= block.nodeRating) && seenBlocks.getIfPresent(block.hash) == null
+        return (curr == null || curr.nodeRating <= block.nodeRating) && seenBlocks.getIfPresent(block.hash) == null &&
+            block.height >= ((curr?.height ?: 0) - 10)
```

Comment:
Now if `curr` is null, the check will look like this: `block.height >= -10`.

Maybe something like this?
```kotlin
block.height >= (curr?.height?.minus(10) ?: 0)
```

Reply:
Yea, it's a good idea. Done